sim/src/core/EntityManager.cpp

```diff
@@ -155,6 +155,18 @@ namespace mars {
       return 0;
     }
 
+
+    std::vector<SimEntity*> EntityManager::getEntities(const std::string &name) {
+      std::vector<SimEntity*> out;
+      for (std::map<unsigned long, SimEntity*>::iterator iter = entities.begin();
+          iter != entities.end(); ++iter) {
+        if (iter->second->getName().find(name) != std::string::npos) {
```

Comment:
Here you could also use `utils::matchPattern(pattern, string)` see:
http://rock-simulation.github.io/mars/d3/da5/namespacemars_1_1utils.html#af9e2091c1fce3985e2e6d63d0e317338